src/Psl/DateTime/Timezone.php

```diff
@@ -0,0 +1,546 @@
+    case PacificChuuk = 'Pacific/Chuuk';
+    case PacificEaster = 'Pacific/Easter';
+    case PacificEfate = 'Pacific/Efate';
+    case PacificEnderbury = 'Pacific/Enderbury';
+    case PacificFakaofo = 'Pacific/Fakaofo';
+    case PacificFiji = 'Pacific/Fiji';
+    case PacificFunafuti = 'Pacific/Funafuti';
+    case PacificGalapagos = 'Pacific/Galapagos';
+    case PacificGambier = 'Pacific/Gambier';
+    case PacificGuadalcanal = 'Pacific/Guadalcanal';
+    case PacificGuam = 'Pacific/Guam';
+    case PacificHonolulu = 'Pacific/Honolulu';
+    case PacificKiritimati = 'Pacific/Kiritimati';
+    case PacificKosrae = 'Pacific/Kosrae';
+    case PacificKwajalein = 'Pacific/Kwajalein';
+    case PacificMajuro = 'Pacific/Majuro';
+    case PacificMarquesas = 'Pacific/Marquesas';
+    case PacificMidway = 'Pacific/Midway';
+    case PacificNauru = 'Pacific/Nauru';
+    case PacificNiue = 'Pacific/Niue';
+    case PacificNorfolk = 'Pacific/Norfolk';
+    case PacificNoumea = 'Pacific/Noumea';
+    case PacificPago_Pago = 'Pacific/Pago_Pago';
+    case PacificPalau = 'Pacific/Palau';
+    case PacificPitcairn = 'Pacific/Pitcairn';
+    case PacificPohnpei = 'Pacific/Pohnpei';
+    case PacificPort_Moresby = 'Pacific/Port_Moresby';
+    case PacificRarotonga = 'Pacific/Rarotonga';
+    case PacificSaipan = 'Pacific/Saipan';
+    case PacificTahiti = 'Pacific/Tahiti';
+    case PacificTarawa = 'Pacific/Tarawa';
+    case PacificTongatapu = 'Pacific/Tongatapu';
+    case PacificWake = 'Pacific/Wake';
+    case PacificWallis = 'Pacific/Wallis';
+
+    /**
+     * Returns the default system time zone as a {@see Timezone} enum instance.
+     *
+     * This method attempts to determine the current default time zone set in the system and return
+     * its corresponding {@see Timezone} enum instance. If the system's default time zone does not match any
+     * of the enum cases, UTC is returned as the default.
+     *
+     * @return Timezone The system's default time zone as a {@see Timezone} enum instance or UTC if not found.
+     */
+    public static function default(): Timezone
+    {
+        return self::tryFrom(date_default_timezone_get()) ?? self::UTC;
+    }
+
+    /**
+     * Calculates the time zone offset for a given {@see TemporalInterface} instance.
+     *
+     * This method determines the UTC offset for the time zone, based on the specific date and time
+     * represented by the provided {@see TemporalInterface} instance. This is important because the offset
+     * can vary due to daylight saving time changes.
+     *
+     * @param TemporalInterface $temporal The temporal object to calculate the time zone offset for.
+     *
+     * @return Duration The offset from UTC as a Duration instance.
+     *
+     * @mutation-free
+     */
+    public function getOffset(TemporalInterface $temporal): Duration
+    {
+        return Internal\zone_override($this, static function () use ($temporal): Duration {
```

Comment:
I have the feeling that this one is being used as a hack here.
Wouldn't it be less hacky to use `DateTimeZone::getOffset()` instead?